pkg/ingester/ingester.go

```diff
@@ -1491,9 +1493,9 @@ func (i *Ingester) createTSDB(userID string) (*userTSDB, error) {
 		IsolationDisabled:              !i.cfg.BlocksStorageConfig.TSDB.IsolationEnabled,
 		HeadChunksWriteQueueSize:       i.cfg.BlocksStorageConfig.TSDB.HeadChunksWriteQueueSize,
 		NewChunkDiskMapper:             i.cfg.BlocksStorageConfig.TSDB.NewChunkDiskMapper,
-		AllowOverlappingQueries:        i.cfg.BlocksStorageConfig.TSDB.OOOAllowance > 0, // always true if out of order support is enabled
-		AllowOverlappingCompaction:     false,                                           // always false since Mimir only uploads lvl 1 compacted blocks
-		OOOAllowance:                   int64(i.cfg.BlocksStorageConfig.TSDB.OOOAllowance / time.Millisecond),
+		AllowOverlappingQueries:        oooAllowance > 0, // true if out of order support is enabled
+		AllowOverlappingCompaction:     false,            // always false since Mimir only uploads lvl 1 compacted blocks
+		OOOAllowance:                   int64(oooAllowance / time.Millisecond),
```

Comment:
It's not clear from `OOOAllowance` documentation what unit it is using. Is it milliseconds? Can we use `oooAllowance.Milliseconds()` instead?

Reply:
The definition is same as that for BlockDuration config(s) in TSDB and any other time configuration. For TSDB it is just a int64 and everything is a relative measure. We the users of TSDB have it in milliseconds. Will make it .Milliseconds().